### PR TITLE
fix: clarify KMS/KP and endpoints in solution/standard

### PR DESCRIPTION
**Pull request by shemau**
### Description

Ref: #441, possibly also  #446

Clarify which inputs/outputs are key protect specific, versus kms generic.
Changes impacted service endpoints, so fully implemented allowed networks/endpoint type.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Review by ocofaigh — CHANGES_REQUESTED** [review]
`service_endpoints` is used in the tests so needs to be updated now to `key_protect_allowed_network`

File: outputs.tf [hunk 1]
```diff
@@ -17,6 +17,12 @@ output "key_protect_name" {
   value       = length(module.key_protect) > 0 ? module.key_protect[0].key_protect_name : null
 }
 
+# The ID is the CRN, output ID as CRN until the base module is updated
+output "key_protect_crn" {
+  description = "Key Protect service instance CRN when an instance is created, otherwise null"
+  value       = can(module.key_protect[0].key_protect_id) ? module.key_protect[0].key_protect_id : null
```
Comment (ocofaigh) [inline outputs.tf:23]: this should use the `key_protect_crn` output coming in https://github.com/terraform-ibm-modules/terraform-ibm-key-protect/pull/555

**Review by ocofaigh — CHANGES_REQUESTED** [review]
You will need to wait for https://github.com/terraform-ibm-modules/terraform-ibm-key-protect/pull/555 to be merged and new release ready so you can bump the version in this PR and add the `key_protect_crn` output. Your also going to have to skip the upgrade test due to variable renaming, but thats ok since the DA is not public

**Comment by ocofaigh** [comment]
@shemau https://github.com/terraform-ibm-modules/terraform-ibm-key-protect/releases/tag/v2.7.0 is now released

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@shemau please skip upgrade test since you renamed the var. Its fine to skip since this DA is not yet public

**Comment by shemau** [comment]
The Schematics DA Upgrade test is failing because the inputs to the module change. Specifically,  `"service_endpoints":"public-and-private",` becomes `"kms_endpoint_type":"public",`. 

I manually ran a Schematics DA plan on the main branch and the PR branch. The resulting plan was identical, despite the different inputs. So there is no issue during upgrade when the calling code has been update to pass kms_endpoint_type=public where it previously passed service_endpoints=public-and-private.

Skipping the upgrade test is appropriate.

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**